### ISSUE TITLE
Add WebUI interface for emergency cluster repair

### ISSUE THREAD

**Issue by deontologician** [rethinkdb/rethinkdb#4392]
This would probably involve adding two levels of "repair" to the `table_availability` issue box.

1. If we still have a quorum and there's no data loss, we can suggest reconfiguring to fix the issue
2. According to rethinkdb/docs#787 the `emergency_repair` command distinguishes between different levels of repair needed. If it doesn't require an emergency repair it raises an exception. If possible, we should hook into this and avoid suggesting emergency repairs when it isn't required.

The interface would work similarly to the reconfigure interface, where it executes a dry-run and when the user explicitly accepts the configuration it commits it. Obviously, in emergency repair mode, it would have a lot more explicit warnings and explanations of exactly what the consequences would be.

**Comment by deontologician** [comment]
@timmaxw is it possible to currently get the emergency_repair category for the shards of a table? That would help in building an interface on top of it. Or, indirectly, can those categories be reliably  regenerated by looking at table_status?

**Comment by timmaxw** [comment]
It's not possible to get the emergency_repair category for each shard; nor is it possible to determine them from `table_status`.

I suggest that we don't make a web UI interface for `emergency_repair` in the first release.

**Comment by deontologician** [comment]
Yeah, this is definitely not for 2.1, but I wanted to open the issue so we can think about it

**Comment by danielmewes** [comment]
I think this is actually fairly important from a user experience point of view.
Will try to schedule this for 2.3.

**Comment by danielmewes** [comment]
This change https://github.com/rethinkdb/rethinkdb/issues/5267 makes emergency repair easier to discover at least. We should still try to implement a graphical interface for it.
